### ISSUE TITLE
Declare entry point in `setup.py` for `cupy.array_api`

### ISSUE THREAD

**Issue by kmaehashi** [cupy/cupy#6219]
### Description

c.f. https://github.com/numpy/numpy/pull/19800

**Comment by leofang** [comment]
Good catch, I missed that the NumPy PR was merged. The entry point is an optional feature (https://github.com/data-apis/array-api/pull/297) but since NumPy has it, it'd be nice for us to follow.

**Comment by kmaehashi** [comment]
Closing as `cupy.array_api` module is going to sunset.
